pkg/network/go/bininspect/utils.go

```diff
@@ -170,25 +134,20 @@ func SymbolToOffset(f *elf.File, symbol string) (uint32, error) {
 
 	var executableSection *elf.Section
 
-	for symbolName, symbolData := range syms {
-		if symbolName == symbol {
-			// Find what section the symbol is in by checking the executable section's
-			// addr space.
-			for m := range sectionsToSearchForSymbol {
-				if symbolData.Value > sectionsToSearchForSymbol[m].Addr &&
-					symbolData.Value < sectionsToSearchForSymbol[m].Addr+sectionsToSearchForSymbol[m].Size {
-					executableSection = sectionsToSearchForSymbol[m]
-					break
-				}
-			}
-
-			if executableSection == nil {
-				return 0, errors.New("could not find symbol in executable sections of binary")
-			}
-
-			return uint32(symbolData.Value - executableSection.Addr + executableSection.Offset), nil
+	// Find what section the symbol is in by checking the executable section's
+	// addr space.
+	for m := range sectionsToSearchForSymbol {
+		if symbol.Value > sectionsToSearchForSymbol[m].Addr &&
+			symbol.Value < sectionsToSearchForSymbol[m].Addr+sectionsToSearchForSymbol[m].Size {
```

Comment:
Nitpick, but I think adding intermediate variables would improve readability here:
```suggestion
                sectionStart := sectionsToSearchForSymbols[m].Addr
                sectionEnd := sectionStart + sectionsToSearchForSymbols[m].Size
		if symbol.Value > sectionStart && symbol.Value < sectionEnd {
```
Also, shouldn't we consider the symbol is in the section if `symbol.Value == sectionStart` meaning the loop condition would be `symbol.Value >= sectionStart && symbol.Value < sectionEnd`?